src/test_l3_rmfAudioCapture.c

```diff
@@ -0,0 +1,1062 @@
+    fwrite(&data_rate, 4, 1, file);
+    fwrite(&block_align, 2, 1, file);
+    fwrite(&bits_per_sample, 2, 1, file);
+    fwrite("data", 1, 4, file);
+    fwrite(&ctx_data->bytes_received, 4, 1, file);
+
+    /* Write PCM Data */
+    fwrite(ctx_data->data_buffer, 1, ctx_data->bytes_received, file);
+
+    fclose(file);
+    if(ctx_data->data_buffer) 
+    {
+        free(ctx_data->data_buffer);
+        ctx_data->data_buffer = NULL;
+    }
+    UT_LOG_INFO("test_l3_write_wav_file created output file : %s", filename);
+    return RMF_SUCCESS;
+}
+
+/**
+ * @brief Function to monitor bytes received to detect jitter
+ *
+ * This function checks byte received in intervals to make sure there is no jitter
+ */
+static void* monitorBufferCount(void* context_blob) 
+{
+    UT_LOG_INFO("Created thread to monitor buffer for jitter");
+    RMF_audio_capture_struct *ctx_data = (RMF_audio_capture_struct *)context_blob;
+
+    uint32_t bytes_received = 0;
+    uint32_t difference_in_bytes = 0;
+    time_t start_time = time(NULL);
+    time_t end_time = start_time + ctx_data->jitter_test_duration;
+
+    rmf_Error *result = malloc (sizeof(rmf_Error));
+    if (result == NULL)
+    {
+        UT_LOG_ERROR ("malloc for storing rmf_Error failed, refer prints to confirm if jitter test passed");
+    }
+
+    while ((ctx_data->cookie == 1) && (time(NULL) < end_time))
+    {
+        difference_in_bytes = ctx_data->bytes_received - bytes_received;
+        if (difference_in_bytes < ctx_data->jitter_threshold) 
+        {
+            UT_LOG_INFO ("Bytes received in last iteration : %d. This is less than threshold level of %d bytes.\n", difference_in_bytes, ctx_data->jitter_threshold);
+            UT_LOG_ERROR ("Jitter detected !");
+            *result = RMF_ERROR;
+            return (void *)result;
+        }
+        bytes_received = ctx_data->bytes_received;
+        usleep (ctx_data->jitter_monitor_sleep_interval);
+    }
+    UT_LOG_INFO("No jitter detected");
+    *result = RMF_SUCCESS;
+    return (void *)result;
+}
+
+/**
+ * @brief Function to choose if test steps are for Primary/Auxiliary audio capture
+ *
+ * This function is in functions that require user to choose audio capture type.
+ * Menu is displayed only when auxiliary capture is supported as per yaml.
+ */
+static int getAudioCaptureType(void)
+{
+    int32_t choice = 1;
+    if (true == g_aux_capture_supported)
+    {
+        UT_LOG_MENU_INFO("------------------------------------------");
+        UT_LOG_MENU_INFO("Supported RMF audio capture types are:");
+        UT_LOG_MENU_INFO("------------------------------------------");
+        UT_LOG_MENU_INFO("\t#   %-20s","Supported capture type");
+        UT_LOG_MENU_INFO("\t1.  %-20s","PRIMARY");
+        UT_LOG_MENU_INFO("\t2.  %-20s","AUXILIARY");
+        UT_LOG_MENU_INFO("------------------------------------------");
+        UT_LOG_MENU_INFO(" Select the audio capture type :");
+        scanf("%d", &choice);
```

Comment:
could you please move scanf and readAndDiscardRestOfLine one common function.

Reply:
Updated in https://github.com/rdkcentral/rdk-halif-test-rmf_audio_capture/pull/21, that PR for c-tests.

Reply:
could you please update on this comment?